server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -397,6 +330,48 @@ public void run() {
             }
         }
 
+        /**
+         * This method evaluates the health info of this node and if there is a change it sends an update request to the health node.
+         *
+         * @return whether the next run has been scheduled.
+         */
+        private boolean sendUpdate(Runnable scheduleNextRun) {
```

Comment:
I understand that this method helps with the nesting, but I find it a bit less helpful that it splits the logic of scheduling the next run which is captured on the surrounding try-catch. 

What do you think, should we see if we can separate it a bit differently?